### DISCUSSION TITLE
Even though I've specified method: "POST" node-fetch still uses get

### DISCUSSION THREAD

**Discussion by flamekiller22** [node-fetch/node-fetch#1671] [Q&A] [answered]
This is the code I'm using:

```js
async authorize(credentials, req) {
        const payload = {
          email: credentials.email,
          password: credentials.password
        }

        const res = await fetch(`${process.env.API_URL}auth/login`, {
          method: 'post',
          body: JSON.stringify(payload),
          headers: {
            'Content-Type': 'application/json'
          }
        });

        const user = await res.json()

        if (!res.ok || user?.method === "GET") {
          console.log('Auth response:', user);
          throw new Error('Invalid Username & Password combination!');
        }

        if (res.ok && user) {
          return user;
        }

        console.log('Auth response:', user);
        return null
      }
```
      
      this is the response I get:
      NOTE: the api responds with { method: "GET" } if using get.
      
      `Auth response: { method : 'GET' }`

**Comment by flamekiller22** [comment]
And yes I tried using `method: "POST"` rather than `method: "post"`

**Comment by flamekiller22** [comment, marked as answer]
Found the problem.
Turns out when following a redirect it does not honour the method and just uses GET.
Not sure if this is expected behaviour or a bug so leaving this in Q&A for now.

**Reply by GabenGar** [reply]
This depends on the status code of the redirect. `301` and `302` will switch to `GET`, while `307` and `308` will preserve the method. So it's a server configuration issue.
